src/conv.rs

```diff
@@ -737,6 +739,7 @@ where
     fn from_boxed_sample_slice(slice: Box<[S]>) -> Option<Self>;
 }
 
+#[cfg(feature = "frame")]
```

Comment:
We can get rid of this (and similar) `#[cfg]` by moving this trait to the `frame` module. This would be a breaking change, however. @mitchmindtree, can you let me know what you think about this?